native/src/fairseq2n/data/data_pipeline.cc

```diff
@@ -391,6 +392,27 @@ data_pipeline_builder::bucket_by_length(
     return std::move(*this);
 }
 
+data_pipeline_builder
+data_pipeline_builder::dynamic_bucket(
+    float64 threshold,
+    cost_fn fn,
+    std::optional<std::size_t> maybe_nb_min,
+    std::optional<std::size_t> maybe_nb_max,
+    bool drop_remainder) &&
+{
+    if (threshold <= 0)
+        throw_<std::invalid_argument>("`threshold` must be greater than zero.");
+    if (maybe_nb_max && *maybe_nb_max == 0)
+        throw_<std::invalid_argument>("`nb_max` must be greater than zero.");
+
```

Comment:
do you check somewhere that `nb_max>=nb_min` ?

Reply:
i'll add this check